### ISSUE TITLE
Should ActiveDataProvider extends [[BaseDataProvider::refresh()]] to clear $_queryResults?

### ISSUE THREAD

**Issue by sizeg** [yiisoft/yii2-elasticsearch#124]
### What steps will reproduce the problem?

```php
$dataProvider = new \yii\elasticsearch\ActiveDataProvider([
    'query' => $someQuery,
]);
$dataProvider->getTotalCount(); // for example 65

// Create new query and set to the same dataprovider
$dataProvider->query = $newQuery;
$dataProvider->refresh();
$dataProvider->getTotalCount(); // again 65, expected for example 95
```

### What is expected

Renew query results after calling `refresh()` method.

### Decision

I suggest to add this lines to \yii\elasticsearch\ActiveDataProvider:

```php
/**
 * @inehritdoc
 */
public function refresh()
{
    parent::refresh();
    $this->_queryResults = null;
}
```

I can prepare a pull request for it.

**Comment by beowulfenator** [comment]
Go ahead and make the PR. Thanks!

**Comment by beowulfenator** [comment]
Fixed in 2.0.5 and merged into master
